src/components/pages/group/form/speciesCheckBox/index.tsx

```diff
@@ -0,0 +1,167 @@
+      languageId: 0,
+      spacialCOverage: group.sGroup ? group.sGroup : [],
+      habitatCoverage: group.habitatCoverage ? group.habitatCoverage : [],
+      sendDigestMail: true,
+      tags: group.tags,
+      invitationData: {
+        userGroupId: 0,
+        founderIds: founder.idsList,
+        moderatorsIds: moderator.idsList,
+        founderEmail: founder.emailList,
+        moderatorsEmail: moderator.emailList
+      }
+    };
+    // console.log("the redo you",hform.getValues())
+    emit(SYNC_SINGLE_GROUP, { group: payload, instant: true });
+    onClose();
+  };
+  return isOpen ? (
+    <Box mb={8} minH="calc(100vh - var(--heading-height))">
+      <PageHeading>{t("GROUP.CREATE_GROUP_TITLE")}</PageHeading>
+      <Box className="white-box" style={{ padding: "10px 30px", margin: "10px" }}>
+        <h2 className="mt4" style={{ margin: "20px 0px", fontSize: 25 }}>
+          <strong>Core Elements</strong>
+        </h2>
+        <form onSubmit={handleFormSubmit}>
+          <TextBoxField
+            name="group_name"
+            isRequired={true}
+            label={t("GROUP.GROUP_NAME")}
+            form={hform}
+          />
+          <TextAreaField
+            name="group_description"
+            label={t("GROUP.GROUP_DESCRIPTION")}
+            form={hform}
+          />
+          <GroupIconUploader form={hform} name="group_icon" />
+          <GroupSelector
+            name="sGroup"
+            label={t("GROUP.SPECIES_GROUP_COVERAGE")}
+            options={speciesGroups}
+            form={hform}
+          />
+          <GroupSelector
+            name="habitatCoverage"
+            label={t("GROUP.HABITATS_COVERED")}
+            options={habitats}
+            form={hform}
+          />
+          <RadioInputField
+            form={hform}
+            isInline={false}
+            name="group_invitaion"
+            label="Can user join the group without invitaion"
+            options={[
+              { label: "yes", value: "true" },
+              { label: "no", value: "false" }
+            ]}
+          />
+          <MapInputForm form={hform} />
+          <SelectAsync
+            name="tags"
+            label={t("OBSERVATION.TAGS")}
+            hint={t("OBSERVATION.TAGS_HINT")}
+            form={hform}
+            multiple={true}
+            onQuery={onTagsQuery}
+            mb={2}
+          />
+          <div className="white-box">
+            <AdminInviteField
+              adminList={adminInviteList}
+              form={hform}
+              adminTitle={"GROUP.GROUP_ADMIN_TITLE"}
+            />
+            <Button
+              className="white-box-padding"
+              type="submit"
+              onSubmit={handleFormSubmit}
+              variantColor="green"
+            >
+              Button
+            </Button>
+          </div>
+        </form>
+      </Box>
+    </Box>
+  ) : (
+    <SavingGroup />
```

Comment:
Use submit button's loading state instead of hiding entire form because this might be better user experiance in this scenario